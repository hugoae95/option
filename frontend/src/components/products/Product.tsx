import "./Product.scss";
import parse from "html-react-parser";
import { Grid } from "@mui/material";
import { IProduct } from "../../types/products.types";

interface IProps {
  product: IProduct;
}
export const Product = (props: IProps) => {
  const { id, title, image, sku, price, barcode, description } = props.product;
  return (
    <div key={id} className="itemsCarousel item-cause">
      <Grid container className="donor-items blog-items" spacing={2}>
        <Grid item xs={12}>
          <img src={image} alt={title} title={title} />
        </Grid>
        <Grid item xs={12}>
          <h4>{title}</h4>
        </Grid>
      </Grid>
      <div className="description_cause">{parse(description)}</div>
      <div className="price_cause">Price: {price}</div>
    </div>
  );
};
