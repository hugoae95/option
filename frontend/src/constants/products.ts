export const UPDATE_PRODUCTS= "UPDATE_PRODUCTS";
export const RESET_PRODUCTS = "RESET_PRODUCTS";
export const SEARCH_PRODUCTS = "SEARCH_PRODUCTS";