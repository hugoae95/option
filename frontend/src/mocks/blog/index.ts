export const blog = [
    {
        title: 'Las 3 grandes problemáticas sociales de Colombia',
        image: 'https://www.lifeder.com/wp-content/uploads/2020/02/protesta-colombia-manifestacion-colectivo-enfado-derechos.jpg',
        description: '<p>Colombia es un país que como la mayoría de países de América Latina, enfrenta múltiples problemas sociales que tienen una incidencia negativa en el progreso desde perspectivas económicas, políticas y ambientales, entre otras.</p>',
        link: 'https://sjrcolombia.org/los-problemas-sociales-de-colombia/',
    },
    {
        title: 'Las 3 grandes problemáticas sociales de Colombia',
        image: 'https://www.lifeder.com/wp-content/uploads/2020/02/protesta-colombia-manifestacion-colectivo-enfado-derechos.jpg',
        description: '<p>Colombia es un país que como la mayoría de países de América Latina, enfrenta múltiples problemas sociales que tienen una incidencia negativa en el progreso desde perspectivas económicas, políticas y ambientales, entre otras.</p>',
        link: 'https://sjrcolombia.org/los-problemas-sociales-de-colombia/',
    },
    {
        title: 'Las 3 grandes problemáticas sociales de Colombia',
        image: 'https://www.lifeder.com/wp-content/uploads/2020/02/protesta-colombia-manifestacion-colectivo-enfado-derechos.jpg',
        description: '<p>Colombia es un país que como la mayoría de países de América Latina, enfrenta múltiples problemas sociales que tienen una incidencia negativa en el progreso desde perspectivas económicas, políticas y ambientales, entre otras.</p>',
        link: 'https://sjrcolombia.org/los-problemas-sociales-de-colombia/',
    },
    {
        title: 'Las 3 grandes problemáticas sociales de Colombia',
        image: 'https://www.lifeder.com/wp-content/uploads/2020/02/protesta-colombia-manifestacion-colectivo-enfado-derechos.jpg',
        description: '<p>Colombia es un país que como la mayoría de países de América Latina, enfrenta múltiples problemas sociales que tienen una incidencia negativa en el progreso desde perspectivas económicas, políticas y ambientales, entre otras.</p>',
        link: 'https://sjrcolombia.org/los-problemas-sociales-de-colombia/',
    },
    {
        title: 'Las 3 grandes problemáticas sociales de Colombia',
        image: 'https://www.lifeder.com/wp-content/uploads/2020/02/protesta-colombia-manifestacion-colectivo-enfado-derechos.jpg',
        description: '<p>Colombia es un país que como la mayoría de países de América Latina, enfrenta múltiples problemas sociales que tienen una incidencia negativa en el progreso desde perspectivas económicas, políticas y ambientales, entre otras.</p>',
        link: 'https://sjrcolombia.org/los-problemas-sociales-de-colombia/',
    },
];