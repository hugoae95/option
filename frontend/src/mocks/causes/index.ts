export const causes = [
    {
        title: 'Por los niños del Chocó Por los niños del Chocó Por los niños del Chocó Por los niños del Chocó',
        image: 'https://www.unicef.org/lac/sites/unicef.org.lac/files/styles/hero_desktop/public/%C2%A9-UNICEF_UNI235471_Willocq-1.jpg?itok=3xN2lCY7',
        foundation: 'Organización de las Naciones Unidas',
        image_foundation: 'https://www.undp.org/sites/g/files/zskgke326/files/migration/co/UNDP_Co_Logo_ONU_Horizontal.png',
        description: '<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pulvinar scelerisque aenean tincidunt massa in interdum et senectus. Sollicitudin tortor et at sit. In et malesuada adipiscing sed eget commodo mattis. Feugiat viverra ac cras nunc neque auctor lacinia pretium in.</p>',
        donors: 47,
    },
    {
        title: 'Por los niños del Chocó 1',
        image: 'https://www.unicef.org/lac/sites/unicef.org.lac/files/styles/hero_desktop/public/%C2%A9-UNICEF_UNI235471_Willocq-1.jpg?itok=3xN2lCY7',
        foundation: 'Organización de las Naciones Unidas',
        image_foundation: 'https://www.undp.org/sites/g/files/zskgke326/files/migration/co/UNDP_Co_Logo_ONU_Horizontal.png',
        description: '<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pulvinar scelerisque aenean tincidunt massa in interdum et senectus. Sollicitudin tortor et at sit. In et malesuada adipiscing sed eget commodo mattis. Feugiat viverra ac cras nunc neque auctor lacinia pretium in.</p>',
        donors: 48,
    },
    {
        title: 'Por los niños del Chocó 2',
        image: 'https://www.unicef.org/lac/sites/unicef.org.lac/files/styles/hero_desktop/public/%C2%A9-UNICEF_UNI235471_Willocq-1.jpg?itok=3xN2lCY7',
        foundation: 'Organización de las Naciones Unidas',
        image_foundation: 'https://www.undp.org/sites/g/files/zskgke326/files/migration/co/UNDP_Co_Logo_ONU_Horizontal.png',
        description: '<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pulvinar scelerisque aenean tincidunt massa in interdum et senectus. Sollicitudin tortor et at sit. In et malesuada adipiscing sed eget commodo mattis. Feugiat viverra ac cras nunc neque auctor lacinia pretium in.</p>',
        donors: 49,
    },
    {
        title: 'Por los niños del Chocó 3',
        image: 'https://www.unicef.org/lac/sites/unicef.org.lac/files/styles/hero_desktop/public/%C2%A9-UNICEF_UNI235471_Willocq-1.jpg?itok=3xN2lCY7',
        foundation: 'Organización de las Naciones Unidas',
        image_foundation: 'https://www.undp.org/sites/g/files/zskgke326/files/migration/co/UNDP_Co_Logo_ONU_Horizontal.png',
        description: '<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pulvinar scelerisque aenean tincidunt massa in interdum et senectus. Sollicitudin tortor et at sit. In et malesuada adipiscing sed eget commodo mattis. Feugiat viverra ac cras nunc neque auctor lacinia pretium in.</p>',
        donors: 50,
    }
]