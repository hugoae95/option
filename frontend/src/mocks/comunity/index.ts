export const comunities = [
    {   
        id: '1',
        title: 'Laura Trujillo',
        job: 'CFO, Restaurante Nirvana',
        image: '/assets/svg/people.svg',
        description: '<p>En el mundo existen muchas necesidades y saber que existen iniciativas como wehehe me brinda tranquilidad porque de este modo podemos ayudar de forma segura y sin ninguna limitación.</p>',
        stars: 5,
    },
    {
        id: '2',
        title: 'Laura Trujillo',
        job: 'CFO, Restaurante Nirvana',
        image: '/assets/svg/people.svg',
        description: '<p>En el mundo existen muchas necesidades y saber que existen iniciativas como wehehe me brinda tranquilidad porque de este modo podemos ayudar de forma segura y sin ninguna limitación.</p>',
        stars: 5,
    },
    {
        id: '3',
        title: 'Laura Trujillo',
        job: 'CFO, Restaurante Nirvana',
        image: '/assets/svg/people.svg',
        description: '<p>En el mundo existen muchas necesidades y saber que existen iniciativas como wehehe me brinda tranquilidad porque de este modo podemos ayudar de forma segura y sin ninguna limitación.</p>',
        stars: 5,
    },
    {
        id: '4',
        title: 'Laura Trujillo',
        job: 'CFO, Restaurante Nirvana',
        image: '/assets/svg/people.svg',
        description: '<p>En el mundo existen muchas necesidades y saber que existen iniciativas como wehehe me brinda tranquilidad porque de este modo podemos ayudar de forma segura y sin ninguna limitación.</p>',
        stars: 5,
    },
    {
        id: '5',
        title: 'Laura Trujillo',
        job: 'CFO, Restaurante Nirvana',
        image: '/assets/svg/people.svg',
        description: '<p>En el mundo existen muchas necesidades y saber que existen iniciativas como wehehe me brinda tranquilidad porque de este modo podemos ayudar de forma segura y sin ninguna limitación.</p>',
        stars: 5,
    },
    {
        id: '6',
        title: 'Laura Trujillo',
        job: 'CFO, Restaurante Nirvana',
        image: '/assets/svg/people.svg',
        description: '<p>En el mundo existen muchas necesidades y saber que existen iniciativas como wehehe me brinda tranquilidad porque de este modo podemos ayudar de forma segura y sin ninguna limitación.</p>',
        stars: 5,
    },
]