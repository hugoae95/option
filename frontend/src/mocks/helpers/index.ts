export const helpers = [
    {
        title: 'ufounders',
        image: '/assets/svg/helpers/ufounders.svg',
    },
    {
        title: 'mono',
        image: '/assets/svg/helpers/mono.svg',
    },
    {
        title: 'payu',
        image: '/assets/svg/helpers/payu.svg',
    },
    
    {
        title: 'aws',
        image: '/assets/svg/helpers/aws.svg',
    },
]