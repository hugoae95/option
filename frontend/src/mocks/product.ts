export const product = {
    "id": 8005412094261,
    "title": "Product 4",
    "description": "Test",
    "image": "https://cdn.shopify.com/s/files/1/0679/2177/2853/products/shirt_378037390_1000.jpg?v=1668572895",
    "sku": "00004",
    "price": 20.00,
    "barcode": null
};