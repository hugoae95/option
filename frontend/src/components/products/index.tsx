import { Grid } from "@mui/material";
import { IProduct } from "../../types/products.types";
import { Product } from "./Product";

interface IProps {
  products: IProduct[];
}

export const Products = (props: IProps) => {
  const { products } = props;
  return (
    <Grid container>
      {products.length > 0 ? (
        products.map((product: IProduct, index: number) => {
          return (
            <Grid id={"product-" + index} item lg={3} xs={12} key={index}>
              <Product product={product} />
            </Grid>
          );
        })
      ) : (
        <Grid item xs={12}>
          No se encuentran productos en el momento.
        </Grid>
      )}
    </Grid>
  );
};
