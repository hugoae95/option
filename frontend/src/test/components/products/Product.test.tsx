import React from "react";
import { render, screen } from "@testing-library/react";
import { Product } from "./../../../components/products/Product";
import renderer from "react-test-renderer";
import { product } from "../../../mocks/product";

test("renders title of the product", () => {
  render(<Product product={product} />);
  const title = screen.getByText(/Product 4/i);
  expect(title).toBeInTheDocument();
});

test("snaphot generate ", () => {
  const component = renderer.create(<Product product={product} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
