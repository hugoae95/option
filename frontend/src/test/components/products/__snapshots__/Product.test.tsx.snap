// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snaphot generate  1`] = `
<div
  className="itemsCarousel item-cause"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 donor-items blog-items css-mhc70k-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <img
        alt="Product 4"
        src="https://cdn.shopify.com/s/files/1/0679/2177/2853/products/shirt_378037390_1000.jpg?v=1668572895"
        title="Product 4"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <h4>
        Product 4
      </h4>
    </div>
  </div>
  <div
    className="description_cause"
  >
    Test
  </div>
  <div
    className="price_cause"
  >
    Price: 
    20
  </div>
</div>
`;
